File: CPP/program/Motorcycle.cpp
#pragma once
// Baris ini memastikan bahwa file header hanya disertakan sekali dalam unit kompilasi.

#include <iostream> // Library ini digunakan untuk operasi input/output.
#include <string>   // Library ini digunakan untuk menggunakan tipe data string.
#include "Vehicle.cpp" // Ini adalah file header untuk kelas Vehicle.

using namespace std; // Baris ini menggunakan namespace standar.

// Ini adalah deklarasi dari kelas Motorcycle yang merupakan turunan dari kelas Vehicle.
class Motorcycle : public Vehicle {
    private:
        // Ini adalah anggota data pribadi dari kelas Motorcycle.
        string jenis_motor;
        int Kapasitas_tangki;

    public:
        // Ini adalah konstruktor default dari kelas Motorcycle.
        Motorcycle(){
            this->jenis_motor = "";
            this->Kapasitas_tangki = 0;
        }

        // Ini adalah metode setter untuk mengatur nilai anggota data pribadi.
        void setJenisMotor(string jenis_motor) {
            this->jenis_motor = jenis_motor;
        }

        void setKapasitasTangki(int Kapasitas_tangki) {
            this->Kapasitas_tangki = Kapasitas_tangki;
        }

        // Ini adalah metode getter untuk mendapatkan nilai anggota data pribadi.
        string getJenisMotor() {
            return jenis_motor;
        }

        int getKapasitasTangki() {
            return Kapasitas_tangki;
        }

        // Ini adalah destruktor dari kelas Motorcycle.
        ~Motorcycle(){

        }
};

File: Python/program/Motorcycle.py
# Ini adalah import kelas Vehicle yang diperlukan.
from Vehicle import Vehicle

# Ini adalah deklarasi dari kelas Motorcycle yang merupakan turunan dari kelas Vehicle.
class Motorcycle(Vehicle):
    def __init__(self):
        # Ini adalah konstruktor kelas Motorcycle.
        super().__init__()  # Memanggil konstruktor kelas induk (Vehicle)
        self.jenis_motor = ""  # Inisialisasi jenis_motor dengan string kosong
        self.kapasitas_tangki = 0

    # Ini adalah metode setter untuk mengatur nilai atribut objek.
    def set_jenis_motor(self, jenis_motor):
        self.jenis_motor = jenis_motor

    def set_kapasitas_tangki(self, kapasitas_tangki):
        self.kapasitas_tangki = kapasitas_tangki

    # Ini adalah metode getter untuk mendapatkan nilai atribut objek.
    def get_jenis_motor(self):
        return self.jenis_motor

    def get_kapasitas_tangki(self):
        return self.kapasitas_tangki

